### PR TITLE
Grammer and comprehension edits

### PR DESCRIPTION
**Pull request by jlyonsmith**
The documentation for Swinject is great, and one of it's great
strengths.  I would like to see the use of the library continue to
increase. Hence, this commit

- Fixes a bunch of minor grammar mistakes in the text
- Rewrites a few phrases to aid comprehension
- Fixes some Markdown formatting errors

### CODE REVIEW

**Comment by yoichitgy** [comment]
Thank you @jlyonsmith for reviewing and enhancing the documentation of Swinject. I really appreciate your help and kindness👍 

Most of your edit is great, but I'm going to add comments to ask you to fix some parts. Would you check my inline comments?

File: Documentation/DIContainer.md [hunk 1]
```diff
@@ -1,18 +1,21 @@
 # DI Container
 
-Dependency Injection (DI) Container manages dependencies of your system. First you register the way how the dependencies or actual types should be resolved. Then you use the DI container to get instances whose dependencies or actual types are resolved by the DI container. In Swinject, `Container` class represents the DI container.
+Dependency Injection (DI) Container manages the type dependencies of your system. First, you register the  types that should be resolved, with their dependencies. Then you use the DI container to get instances of those types whose dependencies are automatically resolved by the DI container. In Swinject, the `Container` class represents the DI container.
 
-In the field of Dependency Injection or Inversion of Control, different terms are used or terms are used differently. Swinject uses the following terms.
+The terms Dependency Injection and Inversion of Control are synonymous. We use the former term, or DI for short.  Confusingly, different DI frameworks use different terms to describe the core concepts. Swinject uses the following terms:
```
Comment (yoichitgy) [inline Documentation/DIContainer.md]: Inversion of Control is a concept including Dependency Injection. [Wikipedia](https://en.wikipedia.org/wiki/Dependency_injection) says: _"dependency injection is a software design pattern that implements inversion of control for resolving dependencies"_

Would you fix the wording "synonymous"?

File: Documentation/ObjectScopes.md [hunk 2]
```diff
@@ -9,25 +9,25 @@ container.register(AnimalType.self) { _ in Cat() }
...
+With `ObjectScope.Graph`, an instance is always created, as in `ObjectScope.None`, if you directly call `resolve` method of a container, but instances resolved in factory closures are shared during the resolution of the root instance to construct the object graph.
 
 This scope must be specified, or `inObjectScope` must not be called if you register circular dependencies.
 
 ## Container (aka Singleton)
 
-In `ObjectScope.Container`, an instance provided by a container is shared within the container. In other words, when you resolve the type at the first time, it is created by the container invoking the factory closure. The instance is returned by the container in the succeeding resolution of the type.
+In `ObjectScope.Container`, an instance provided by a container is shared within the container. In other words, when you resolve the type fore the first time, it is created by the container by invoking the factory closure. The same instance is returned by the container in any succeeding resolution of the type.
```
Comment (yoichitgy) [inline Documentation/ObjectScopes.md]: Maybe not "**fore** the first time" but "**for** the first time"?

**Comment by yoichitgy** [comment]
I got only 2 comments⬆️ I'll merge your PR after your update.

You've made the documentation easier to read and understand. Thank you!

File: Documentation/DIContainer.md (hunk 1 above)
Comment (jlyonsmith) [inline Documentation/DIContainer.md]: I rewrote this section again, using the phrase above as well as adding some links to Wikipedia for DI and IoC.

File: Documentation/ObjectScopes.md (hunk 2 above)
Comment (jlyonsmith) [inline Documentation/ObjectScopes.md]: Fixed.

**Comment by yoichitgy** [comment]
I appreciate that you've reviewed whole the documentation! Your work will be helpful for users to learn DI and Swinject🚀

I always welcome your pull requests if you have things left to do.

**Pull request merged**